buf/registry/module/v1beta1/label_service.proto

```diff
@@ -92,6 +95,8 @@ message ListLabelsRequest {
   //   - If a Module is referenced, all Labels for the Module are returned.
   //   - If a Label is referenced, this Label is returned.
   //   - If a Commit is referenced, all Labels for the Commit are returned.
+  //
+  // If an archived Label is referenced but include_archived is false, this will return an error.
```

Comment:
Just remove this comment

Reply:
But this field would be needed to support getting a set of labels that includes both archived and unarchived labels for a given resource ref (e.g. commit) right?

Reply:
I think I see the dilemma here. We kind of want to enforce by default that archived labels can't be used to resolve anything important, so this change is proposing that `ResourceRef` can't reference an archived label.

The consequence of this is that ListLabels cannot return archived labels because the request parameter here is a ResourceRef.

The future use-case @akshayjshah is interested in is being able to display all the labels that ever pointed to that commit, even archived ones. In this future world, we would want either:
1. This method to be able to support returning not only archived labels, but also historical labels (not just the labels currently pointing to a commit). The three things we would need:
  1. Add `include_archived` to include archived labels in response (this can be added later)
  2. Add `include_historical` (or equivalent) to include labels that used to point to this commit (this can be added later)
  3. Either allow ResourceRef to reference archived labels, or use a different type here, potentially a `oneof` ResourceRef and LabelRef. This would be a breaking change from current behavior, so if we want this we need to do it now.
2. Create a new RPC.

If we are ok with creating a new RPC in the future for this use case if we need it, then I think we can leave the API as-is and remove this comment.

Reply:
I think for option 1, if we are using `include_*` fields, which would have backwards-compatible defaults, we can also implement later down the road. I think I'm happy removing this comment and we can revisit this when we need a solution.

Reply:
Chatted with Doria live. I was mixed up about needing (1)(iii). We don't need to change the request to accomplish this use case because we will be requesting a commit id (which is valid) and returning archive labels (which is valid).

This means (1)(i) and (1)(ii) are sufficient, and we can do both of those in the future in a backward compatible way. Resolving this thread.